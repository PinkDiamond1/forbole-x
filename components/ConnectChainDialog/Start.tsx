import {
  Table,
  TableBody,
  TableHead,
  DialogContent,
  TableRow,
  TableCell,
  Typography,
  Button,
  Box,
  Avatar,
  useTheme,
} from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import format from 'date-fns/format'
import LogoLight from '../../assets/images/image_default_light.svg'
import LogoDark from '../../assets/images/image_default_dark.svg'
import useStyles from './styles'
import connectableChains from '../../misc/connectableChains'
import { useGeneralContext } from '../../contexts/GeneralContext'

interface StartProps {
  onConnectClick(): void
  connections: ChainConnection[]
}

const Start: React.FC<StartProps> = ({ onConnectClick, connections }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const { theme } = useGeneralContext()
  const themeStyle = useTheme()

  return (
    <DialogContent className={classes.dialogContent}>
      {connections.length ? (
        <Table style={{ position: 'relative' }}>
          <TableHead>
            <TableRow className={classes.tableRow}>
              <TableCell className={classes.tableCell}>{t('chain account')}</TableCell>
              <TableCell className={classes.tableCell}>{t('creation time')}</TableCell>
              <TableCell className={classes.tableCell}>{t('manage')}</TableCell>
            </TableRow>
          </TableHead>

          <TableBody>
            {connections.map(({ creationTime, externalAddress, chainName }, i) => (
              <TableRow key={externalAddress} className={classes.tableRow}>
                <TableCell className={classes.tableCell}>
                  <Box display="flex">
                    <Avatar
                      src={connectableChains[chainName].image}
                      alt={connectableChains[chainName].name}
                    />
                    <Box>
                      <Typography>{connectableChains[chainName].name}</Typography>
                      <Typography variant="body2" color="textSecondary">
                        {externalAddress}
                      </Typography>
                    </Box>
                  </Box>
                </TableCell>
                <TableCell className={classes.tableCell}>
                  {format(creationTime, 'dd MMM yyyy, HH:mm:ss')}
                </TableCell>
                <TableCell className={classes.tableCell}>
                  <Button>{t('disconnect')}</Button>
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      ) : (
        <Box display="flex" flexDirection="column" alignItems="center" mt={2} mb={6}>
          {theme === 'dark' ? (
            <LogoDark width={themeStyle.spacing(30)} />
          ) : (
            <LogoLight width={themeStyle.spacing(30)} />
          )}
          <Box mt={3} mb={2}>
            <Typography>{t('connect chain description')}</Typography>
          </Box>

          <Button onClick={onConnectClick} variant="contained" color="primary">
            {t('connect chain')}
          </Button>
        </Box>
      )}
    </DialogContent>
  )
}

export default Start
