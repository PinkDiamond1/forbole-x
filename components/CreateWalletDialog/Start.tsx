import {
  ButtonBase,
  Button,
  DialogActions,
  DialogContent,
  DialogContentText,
  Grid,
  Typography,
} from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import useStyles from './styles'

interface StartProps {
  onImportWalletClick(): void
  onCreateWalletClick(): void
}

const Start: React.FC<StartProps> = ({ onImportWalletClick, onCreateWalletClick }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()

  return (
    <>
      <DialogContent className={classes.dialogContent}>
        <DialogContentText>{t('create wallet description')}</DialogContentText>
        <Grid container spacing={5}>
          <Grid item md={6}>
            <ButtonBase className={classes.selectionBox} onClick={onImportWalletClick}>
              <Typography variant="body2" align="center" color="textSecondary">
                {t('i have mnemonic phrase')}
              </Typography>
            </ButtonBase>
          </Grid>
          <Grid item md={6}>
            <ButtonBase className={classes.selectionBox} onClick={onCreateWalletClick}>
              <Typography variant="body2" align="center" color="textSecondary">
                {t('i dont have mnemonic phrase')}
              </Typography>
            </ButtonBase>
          </Grid>
        </Grid>
      </DialogContent>
      <DialogActions>
        <Button className={classes.button} color="secondary">
          {t('what is mnemonic phrase')}
        </Button>
      </DialogActions>
    </>
  )
}

export default Start
