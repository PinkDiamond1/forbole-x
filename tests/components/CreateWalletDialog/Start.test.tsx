import React from 'react'
import renderer from 'react-test-renderer'
import Start from '../../../components/CreateWalletDialog/Start'

const onImportWalletClick = jest.fn()
const onCreateWalletClick = jest.fn()

describe('component: CreateWalletDialog - Start', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Start onImportWalletClick={onImportWalletClick} onCreateWalletClick={onCreateWalletClick} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls onImportWalletClick when first button is clicked', () => {
    const component = renderer.create(
      <Start onImportWalletClick={onImportWalletClick} onCreateWalletClick={onCreateWalletClick} />
    )
    renderer.act(() => {
      component.root.findAllByType('button')[0].props.onClick()
    })
    expect(onImportWalletClick).toBeCalled()
  })
  it('calls onCreateWalletClick when second button is clicked', () => {
    const component = renderer.create(
      <Start onImportWalletClick={onImportWalletClick} onCreateWalletClick={onCreateWalletClick} />
    )
    renderer.act(() => {
      component.root.findAllByType('button')[1].props.onClick()
    })
    expect(onCreateWalletClick).toBeCalled()
  })
})

afterEach(() => {
  jest.clearAllMocks()
})
