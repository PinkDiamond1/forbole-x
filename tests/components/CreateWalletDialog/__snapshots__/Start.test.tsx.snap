// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: CreateWalletDialog - Start renders correctly 1`] = `
Array [
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-6"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      common:create wallet description
    </p>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
      >
        <button
          className="MuiButtonBase-root HookGlobalStyles-selectionBox-4"
          disabled={false}
          onBlur={[Function]}
          onClick={[MockFunction]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
          >
            common:i have mnemonic phrase
          </p>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6"
      >
        <button
          className="MuiButtonBase-root HookGlobalStyles-selectionBox-4"
          disabled={false}
          onBlur={[Function]}
          onClick={[MockFunction]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <p
            className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
          >
            common:i dont have mnemonic phrase
          </p>
        </button>
      </div>
    </div>
  </div>,
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text HookGlobalStyles-button-1 MuiButton-textSecondary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:what is mnemonic phrase
      </span>
    </button>
  </div>,
]
`;
